### DISCUSSION TITLE
Using zsh as shell with tmux and vim-slime is slow (the text gets pasted in slower)

### DISCUSSION THREAD

**Discussion by emilBeBri** [jpalardy/vim-slime#391] [Q&A]
So I normally use bash, but would like to switch to zsh. I often have a tmux pane running in a terminal, but I notice that setting 

set-option -g default-shell /bin/zsh

in tmux.conf and then using tmux makes sending code from neovim to tmux noticeably slower - with bash, it is instantanious, but with zsh, there is a lag. I tried it in both alacritty and kitty, and it is the same. 

I don't see why that would be the case? I can't really switch to zsh before I can figure out a solution here, because although I wouldn't say it is 'slow', it is  noticeable and it kinda disturbs my flow when I was used to 'instantanous'

on manjaro linux  fully updated

**Comment by jpalardy** [comment]
Hi @emilBeBri 

From the tmux man page, it seems like `default-shell` only comes into play when opening up a new window (possibly pane too). It doesn't seem like it would interfere 🤔 

In a normal setup:
- tmux is opened
- the window is split
- vim (or nvim) runs in one page, your target REPL in the other
- hitting `ctrl-c` `ctrl-c` tells vim to send a `tmux` command through the shell

Thoughts:
- what happens if you don't set `default-shell`?
- how much logic is there in your zsh config files? (how long to run? That would certainly hit, multiple times, for each send command)
- does this also happen in (non-neo) vim?
- does this also happen without vim — if you try to run the tmux commands yourself, from the command-line?

Let know what you find ^

**Comment by emilBeBri** [comment]
Thank you for a very thourough attempt at helping! I  should have probably have explained, that I'm not using nvim in tmux - I have one term w. termux, one with nvim.  So I then send code from the nvim (non-tmux) terminal to the tmux pane in the other terminal, and it is here that it near instantaneous with bash, and have a noticeable lag w. zsh.

(The reason I do it like this is I'd rather have my wm (i3) control the window layout as much as possible, but if there is some kind of benefit I'm mising here I'm always prepared to discover new ways of doing things)

Does this make a difference to your suggestions?

Best, Emil

**Reply by jpalardy** [reply]
That's right … it doesn't really matter if you're in tmux with you neovim. (that's just how I use it 😄 )

Most of my other ideas would still apply. Try to send the commands manually and see if that's slow.

If that's not clear, I can give you a few examples.

**Reply by emilBeBri** [reply]
Hmm it seems like now it works fine -  not sure why that is, perhaps there was a huge process running in background zsh or something? anyway it's gone now. Thanks anyway though 

btw just a quick question (if you want I can open open another thread just say so) - is there a way to change the tmux pane I'm sending to, once set? sometimes  I by mistake I send to one pane and it was another I needed, and my current way of fixing this is restarting nvim, and then choosing the right one lol

**Reply by jpalardy** [reply]
Hi @emilBeBri 

It's nice that it _seems_ resolved. Let me know if it happens again.

You can force a "re-config" with <kbd>ctrl</kbd><kbd>c</kbd>, <kbd>v</kbd> 😄 
(if you haven't changed the defaults)

https://github.com/jpalardy/vim-slime/blob/main/plugin/slime.vim#L35

**Comment by emilBeBri** [comment]
Thanks again - works great with the re-config, too :) 

 I'll keep this open for now and will try out your suggestions if it reappears, and will close it in a week or two if it does not.
